core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -162,6 +163,39 @@ public String defaultHostname() {
         }
     }
 
+    /**
+     * Returns the local {@link ServerPort} that this {@link Server} is listening to.
+     *
+     * @throws IllegalStateException if there is no active local port available
+     *                               or the server is not started yet
+     */
+    public int activeLocalPort() {
+        return activeLocalPort0(null);
+    }
+
+    /**
+     * Returns the local {@link ServerPort} which serves the given {@link SessionProtocol}.
+     *
+     * @throws IllegalStateException if there is no active local port available for the given
+     *                               {@link SessionProtocol} or the server is not started yet
+     */
+    public int activeLocalPort(SessionProtocol protocol) {
+        return activeLocalPort0(requireNonNull(protocol, "protocol"));
+    }
+
+    private int activeLocalPort0(@Nullable SessionProtocol protocol) {
+        synchronized (activePorts) {
+            return activePorts.values().stream()
+                              .filter(activePort -> isLocalPort(activePort, protocol))
+                              .findFirst()
+                              .orElseThrow(() -> new IllegalStateException(
+                                      "no active local ports for " + (protocol == null ? "ALL" : protocol) +
```

Comment:
`no active local ports for ALL` sounds somewhat unnatural. How about:

```java
(protocol == null ? "no active local ports: " 
                  : ("no active local ports for " + protocol + ": ")) +
activePorts.values()
```

Reply:
@trustin yeah i changed again. I considered which is more natural any and all :)